Add .github/dependabot.yml

kolyshkin opened the pull request:
Make sure our dependencies and actions stay up to date.

PS I am not entirely sure that keeping all our go.mod dependencies
at their latest released versions is quite the right approach. Maybe
@thaJeztah have something to say about it.

thaJeztah commented:
Yeah, I'm tempted to say limiting it to updating github actions could make sense; it's good to keep the actions up-to-date, but for `go.mod` dependencies, I'd still consider sticking to Go's MVS (minimum version selection); if there's no reason to update the dependency, then we should specify the minimum required version for this module to do its thing (and consumers can pick whatever SemVer compatible version that's higher).

For Go versions, we could use the `stable` and  `oldstable` keywords if we want to reduce maintenance; might be good to setup a scheduled job so that CI runs periodically, even if there were no changes, so test against `oldest supported`, `oldstable`, and `stable` ?
https://github.com/opencontainers/cgroups/blob/9657f5a18b8d60a0f39fbb34d0cb7771e28e6278/.github/workflows/test.yml#L20



I guess _ideally_ we'd have CodeQL check for vulnerabilities in the modules we use, and it _could_ update those dependencies, but it's likely to have false-positives (so setting up a `govulncheck` check would probably be better.

tianon commented:
> For Go versions, we could use the `stable` and `oldstable` keywords if we want to reduce maintenance; might be good to setup a scheduled job so that CI runs periodically, even if there were no changes, so test against `oldest supported`, `oldstable`, and `stable` ?

To that end, you might enjoy https://github.com/opencontainers/image-spec/commit/00584ffd4333ddb846d4852391e2acb32288c172 👀

kolyshkin commented:
I've updated the PR to exclude go.mod updates, and added a govulncheck run
for every PR, the push on main branch, and once a week (with the rest of the checks).

kolyshkin commented:
govulncheck-action is a tad rusty; filed https://github.com/golang/go/issues/73616 and opened a CL to fix: https://go-review.googlesource.com/c/govulncheck-action/+/670318

kolyshkin commented on .github/workflows/test.yml:
```diff
@@ -7,6 +7,9 @@ on:
       - release-*
   pull_request:
+  schedule:
+    # Runs at 00:00 UTC every Monday
+    - cron: '0 0 * * 1'
```
This used to be in crontab(5) man page, but cron is now gone from the distros in favor of systemd.timer so there's no documentation anymore).

kolyshkin commented:
Hmm, I dunno what to do about the new govulncheck warning, as [the vulnerability](https://pkg.go.dev/vuln/GO-2025-3750) is Windows-specific and this package is Linux-specific, so we're definitely not vulnerable.

The only way I see is to use latest go version so that standard library vulnerabilities will be excluded.
